services/predictors/layoutparser/model/predictor.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+from typing import List
+
+from pydantic import BaseModel, BaseSettings, Field
+
+from model.instance import Instance
+from model.prediction import Prediction, Box, BoxGroup
+import mmda.predictors.lp_predictors as lpp
+from mmda.types.annotation import BoxGroup as AnnotationBox
+from mmda.types.document import Document
+from mmda.types.image import Image
+
+
+class PredictorConfig(BaseSettings):
+    """
+    Path to the model weights
+    """
+
+    weights_path = "lp://PrimaLayout/mask_rcnn_R_50_FPN_3x/config"
+
+
+class Predictor:
+    """
+    Initialize model
+    """
+
+    _lp_predictor: lpp.LayoutParserPredictor
+
+    def __init__(self, config: PredictorConfig):
+        """
+        Load weights
+        """
+        self._lp_predictor = lpp.LayoutParserPredictor.from_pretrained(
+            config.weights_path
+        )
+
+    def predict_batch(self, instances: List[Instance]) -> List[Prediction]:
+        predictions = []
+
+        for instance in instances:
+            doc = Document(
+                symbols="", images=[Image.frombase64(i) for i in instance.page_images]
+            )
+            annotations = self._lp_predictor.predict(doc)
+
+            predictions.append(
```

Comment:
Somewhat painful translation from `mmda` model classes to `pydantic` model classes. Maybe this can be streamlined by converting to json first?

Reply:
does something like this https://github.com/allenai/mmda/blob/4d11cb41a8c270119bc06a5fe0e5f9ebb0da03cb/services/parsers/symbolscraper/model/predictor.py#L52
not exist for these set of mmda and pydantic models?

if not, we could add some classmethods to the pydantic classes that take in mmda models and produce pydantic variants. 

Im also okay with this being as is, even though it is painful as you say.

Reply:
We can do this once we modify `lp_predictor` to return a `Document` class with the `BoxGroup`s added, which requires some work on the `Document` class.